iterator/src/main/java/com/iluwatar/iterator/list/TreasureChestItemIterator.java

```diff
@@ -68,7 +67,7 @@ private int findNextIdx() {
       if (type.equals(ItemType.ANY) || items.get(tempIdx).getType().equals(type)) {
         break;
       }
-    }
+    } while(true);
```

Comment:
this will become an inifinite loop. you should do 
`
do {...
} while(!found)
`

Reply:
Thanks for the review!

The query on lgtm identified the loop condition here as confusing, because `found` isn't assigned in the loop. In fact the two `break` statements are the only means of exit.

Reply:
-sorry, I guess the switch from a `while` to `do` is itself a bit confusing. I guess that's just my habit to use a `do` when the condition is `(true)`.